docs/src/app/components/pages/components/app-bar.jsx

```diff
@@ -135,6 +135,29 @@ class AppBarPage extends React.Component {
                 <MenuItem primaryText="Sign out" />
               </IconMenu>
           } />
+          <br />
+
+          <AppBarNew>
+
+            <div data-position="navIconsGroup">
```

Comment:
A < AppBarNavGroup > would look better here IMHO. And then you don't need to search trough all the children to form and order your groups - let the user group and order them. And if you really need to control ordering, you can use flexbox order.